src/modules/fuzzy-search/look-up.repository.ts

```diff
@@ -105,27 +135,107 @@ export class SequelizeLookUpRepository implements LookUpRepository {
-        return { ...base, item: raw.file };
-      } else if (raw.folder) {
-        return { ...base, item: raw.folder };
-      } else {
-        return { ...base };
-      }
-    });
+    return this.transformResult(result);
   }
 
+  async workspaceSearch(
+    userUuid: UserAttributes['uuid'],
+    workspaceUser: WorkspaceAttributes['workspaceUserId'],
+    workspaceId: WorkspaceAttributes['id'],
+    partialName: string,
+    offset: number,
+  ): Promise<LookUpResult> {
+    const partialNameFormatted = partialName.replace(/\s+/g, ' & ');
+
+    const result = await this.model.findAll({
+      attributes: {
+        include: [
+          [
+            Sequelize.literal(
+              'nullif(ts_rank("tokenized_name", to_tsquery(:partialNameFormatted)), 1)',
+            ),
+            'rank',
+          ],
+          [
+            Sequelize.fn(
+              'similarity',
+              Sequelize.col('LookUpModel.name'),
+              partialName,
+            ),
+            'similarity',
+          ],
+          [
+            Sequelize.literal(
+              `CASE WHEN "LookUpModel"."name" = :partialName THEN 1 ELSE 0 END`,
+            ),
+            'exactMatch',
+          ],
+        ],
+      },
+      where: {
+        user_id: workspaceUser,
+        [Op.or]: [
+          Sequelize.literal(
+            `to_tsquery(:partialNameFormatted) @@ "tokenized_name"`,
+          ),
+          Sequelize.where(
+            Sequelize.fn(
+              'similarity',
+              Sequelize.col('LookUpModel.name'),
+              partialName,
+            ),
+            { [Op.gt]: 0.0 },
+          ),
+        ],
+        [Op.and]: [
+          Sequelize.literal(`"workspaceItemUser"."created_by" = :userUuid`),
+          Sequelize.literal(
+            `"workspaceItemUser"."workspace_id" = :workspaceId`,
+          ),
+        ],
+      },
+      order: [
+        [Sequelize.literal('"exactMatch"'), 'DESC'], // Prioritize exact matches
+        [Sequelize.literal('"rank"'), 'DESC'],
+        [Sequelize.literal('"similarity"'), 'DESC'],
+      ],
+      limit: 5,
+      offset: offset,
+      replacements: {
+        partialName,
+        partialNameFormatted,
+        userUuid,
+        workspaceUser,
+        workspaceId,
+      },
+      include: [
+        {
+          model: FileModel,
+          attributes: ['type', 'id', 'size', 'bucket', 'fileId', 'plainName'],
+          as: 'file',
+        },
+        {
+          model: FolderModel,
+          attributes: ['id'],
+          as: 'folder',
+        },
+        {
+          model: WorkspaceItemUserModel,
+          attributes: [],
+          as: 'workspaceItemUser', // Corrected alias
+          required: true,
+          on: {
+            col1: Sequelize.where(
+              Sequelize.col('LookUpModel.item_id'),
+              '=',
+              Sequelize.col('workspaceItemUser.item_id'),
+            ),
+          },
+        },
+      ],
+    });
```

Comment:
Just out of curiosity as I think you might have already tried it. Since you already created the association between `WorkspaceItemUserModel` and `LookUpModel `, doesn't it let you to make the join without custom on?

Reply:
It was weird, I initially used the custom on to bypass the association and that worked great. But out of nowhere it asked for the association to be created. After that I didn't try using a normal on